### DISCUSSION TITLE
Clean-up unused dependencies

### DISCUSSION THREAD

**Discussion by paescuj** [directus/directus#5368] [Feature Requests]
Currently, there are a lot of unused dependencies in all packages (/, app, api, ...).
Those should be cleaned-up for a number of reasons including to reduce the dev size of Directus which around 1.5 GiB at the moment.

To detect unused dependencies, the tool `depcheck` can assist:
`npx depcheck`
But please note that there will be a lot of false-negatives (I don't think depcheck supports workspaces), so you need to cross-check with other tools like `grep`, ...

**Comment by paescuj** [comment]
https://github.com/directus/directus/pull/9545

**Comment by rijkvanzanten** [comment]
> Currently, there are a lot of unused dependencies in all packages (/, app, api, ...).

Do you have specific examples? Be very careful when it comes to tools like de-check, as like you said, the amount of false-negatives is staggering, and deleting a dependency that's used in some edge case you missed will be absolutely devastating 😬

**Reply by paescuj** [reply]
> > Currently, there are a lot of unused dependencies in all packages (/, app, api, ...).

I may have exaggerated a little 😄 There aren't this many...
Most of the unused dependencies are left-overs from earlier code revisions (Vue2 -> Vue3, ...).

> Be very careful when it comes to tools like de-check, as like you said, the amount of false-negatives is staggering, and deleting a dependency that's used in some edge case you missed will be absolutely devastating 😬

I absolutely agree! I've double checked all removed dependencies in #9545.
